hips/hip-00NN.md

```diff
@@ -0,0 +1,101 @@
+---
+hip: "00NN"
+title: "Support Git protocol for installing chart dependencies"
+authors: [ "Jeff Valore (@rally25rs)", "yxxhero <aiopsclub@163.com>", "Dominykas Blyžė <hello@dominykas.com>" ]
+created: "2021-10-05"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+Currently, Helm supports installing dependencies from various registries, however that requires the charts to be published there. There are use cases which are cumbersome when registries are involved, for example:
+
+- private charts for organizations/individuals who do not have the capacity to maintain the infrastructure required to run a private registry
+- testing work-in-progress charts in their umbrella charts
+
+## Motivation
+
+There are existing ways to achieve installation of charts without using a registry, however they are not very user-friendly and require additional tooling.
+
+- A Helm chart repository is effectively an `index.yaml` with links to downloads. Maintaining such a repository does create a burden of scripts and automation (e.g. Github Actions). This is not always feasible for smaller projects. It also does not really offer an obvious and readily available way of testing pre-releases.
```

Comment:
Note: OCI support does alleviate some of these issues with needing to maintain a chart repository. But doesn't really help for e.g the "testing" use case below.

Reply:
Indeed, and Github itself supports OCI (although I haven't tried publishing charts into it...) - it would be great if there was a shared workflow or a standardized Github action to publish Helm charts for maintainers to use...

Reply:
> or a standardized Github action to publish Helm charts for maintainers to use

Good point. I will double check nothing exists, and add an issue on Helm if not.